libafl/src/generators/mod.rs

```diff
@@ -140,23 +123,13 @@ where
     }
 }
 
-impl<S> RandPrintablesGenerator<S> {
-    /// Creates a new [`RandPrintablesGenerator`], generating up to `max_size` random printable characters.
-    ///
-    /// To skip the 0 `max_size` check, create this using [`Self::from_nonzero`] instead.
-    pub fn new(max_size: usize) -> Result<Self, Error> {
-        let Some(max_size) = NonZero::new(max_size) else {
-            return Err(Error::illegal_argument("The max_size may not be 0."));
-        };
-        Ok(Self::from_nonzero(max_size))
-    }
-
+impl RandPrintablesGenerator {
     /// Returns a new [`RandBytesGenerator`], generating up to `max_size` random bytes.
     #[must_use]
-    pub fn from_nonzero(max_size: NonZeroUsize) -> Self {
-        Self {
-            max_size,
-            phantom: PhantomData,
-        }
+    pub fn new(max_size: usize) -> Self {
+        // # Safety
+        // Saturating added 1 so it's always above 0
+        let max_size = unsafe { NonZeroUsize::new(max_size.saturating_add(1)).unwrap_unchecked() };
```

Comment:
We should check for 0 here and return Error::IllegalArgument (or really have max_size be NonZero) - right now specifying a max_size of 0 will result in generated inputs of size 1.

Reply:
Pick one of the two options (or otherwise return 0 sized inputs lol)